packages/aws-cdk-lib/aws-logs/lib/pattern.ts

```diff
@@ -225,6 +246,16 @@ class JSONStringPattern extends JsonPattern {
   }
 }
 
+/**
+ * A regex comparison for JSON patterns
+ */
+class JSONRegexPattern extends JsonPattern {
+  public constructor(jsonField: string, comparison: string, value: string) {
+    // No validation, we assume these are generated by trusted factory functions
+    super(`${jsonField} ${comparison} %${value}%`);
```

Comment:
Wondering if we should keep this is as `${value}` and throw an error if `value` isn't surrounded by `%`? The original issue complained that extra quotes were added and this will add extra `%`.

Personally haven't used log patterns too much so not sure if its more likely that people would pass in value with encapsulated within `%` or not

Reply:
The problem with the quotes was that it didn't allow for %, which is the only other option, denoting regex. Since this adds the regex support with %, I don't think there is a reason to have the version without any surrounding markers. People who want literal matching use the " mark Version and people who want the regex pattern take this new % version.

Reply:
Fair enough